core/sdk/src/client.ts

```diff
@@ -1119,11 +1124,12 @@ export class Client
     async sendChannelMessage(
         streamId: string,
         payload: ChannelMessage,
-        opts: { beforeSendEventHook?: Promise<void> } = {},
+        opts: SendChannelMessageOptions = {},
     ): Promise<{ eventId: string }> {
         const stream = this.stream(streamId)
         check(stream !== undefined, 'stream not found')
         const localId = stream.view.appendLocalEvent(payload, 'sending', this)
+        opts?.onLocalEventAppended?.(localId)
```

Comment:
opts isn't optional, so the ?. after opts shouldn't be required

Reply:
ok, I made the options optional instead, as they should be